### PR TITLE
[SRVCOM-2352] Bump Knative operator to 1.6.1, remove v1alpha1 crds

### PR DESCRIPTION
**Pull request by skonto**
Fixes JIRA #SRVCOM-2352

## Proposed Changes

- Bumps operator to 1.6.1. We can update to 1.8.x in another PR. Doing it this way to stay on the safe side
and utilize work done previously.
- Removes v1alpha1 crds.

### CODE REVIEW

**Review by ReToCode — COMMENTED** [review]
Can we also drop `setWebookStrategyToNone` and
```
	if err := common.CheckMinimumKubeVersion(ctx.Clients.Kube.Discovery(), "1.24.0"); err != nil {
		for _, name := range crds {
			if err := setWebookStrategyToNone(ctx, name); err != nil {
				return err
			}
		}
	}
```
from `serverless.go`?

**Comment by skonto** [comment]
> Can we also drop setWebookStrategyToNone ...

We need that for 4.10.

File: hack/patches/010-reduce-imc-replicas.patch [hunk 1]
```diff
@@ -3,7 +3,7 @@ index 9dbc884f3..221c696ad 100644
 --- a/vendor/knative.dev/operator/pkg/reconciler/common/ha.go
 +++ b/vendor/knative.dev/operator/pkg/reconciler/common/ha.go
 @@ -26,7 +26,7 @@ import (
- 
+
```
[the remark this answers is not in the document]
Comment (skonto) [reply hack/patches/010-reduce-imc-replicas.patch:6]: Not really it was just cherry picking and modifying have to check.

File: openshift-knative-operator/cmd/operator/main.go [hunk 2]
```diff
@@ -45,10 +44,7 @@ func main() {
 }
 
 func newConversionController(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
-	var (
-		v1beta1  = operatorv1beta1.SchemeGroupVersion.Version
-		v1alpha1 = operatorv1alpha1.SchemeGroupVersion.Version
-	)
+	var v1beta1 = operatorv1beta1.SchemeGroupVersion.Version
 
 	return conversion.NewConversionController(ctx,
```
[the remark this answers is not in the document]
Comment (skonto) [reply openshift-knative-operator/cmd/operator/main.go:49]: I keep it as it is upstream at some point you need it for the future migration to v1.

**Comment by nak3** [comment]
LGTM

File: hack/patches/010-reduce-imc-replicas.patch (hunk 1 above)
[the remark this answers is not in the document]
Comment (ReToCode) [reply hack/patches/010-reduce-imc-replicas.patch:6]: I think it was a blank before? Was just not sure if it has to be a change or not.

**Comment by ReToCode** [comment]
> > Can we also drop setWebookStrategyToNone ...
> 
> We need that for 4.10.

But why is that? The CRD already has `None` and we do no longer downgrade to alpha, why do we need to set it to `None` in code?

**Comment by skonto** [comment]
> But why is that? The CRD already has None and we do no longer downgrade to alpha, why do we need to set it to None in code?

OLM fills in the webhook configuration in the crds, changes strategy from NONE to the actuall webhook that serves the crds (its in the csv).  Here is a sample:

```
  conversion:
    strategy: Webhook
    webhook:
      clientConfig:
        caBundle: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJ2VENDQVdLZ0F3SUJBZ0lJUHE4Y0lrckpzSlV3Q2dZSUtvWkl6ajBFQXdJd1FqRVd>
        service:
          name: knative-operator-webhook-service
          namespace: openshift-serverless
          path: /resource-conversion
          port: 8443
      conversionReviewVersions:
      - v1beta1

```

When we uninstall S-O it still points to the old setup (bug described in https://access.redhat.com/solutions/6992396), so we want to make it empty before we do the upgrade.

**Comment by ReToCode** [comment]
Ok, thanks for clarifying. LGTM overall, holding for https://github.com/openshift-knative/serverless-operator/pull/2017#discussion_r1146196194. WDYT @mgencur?

**Comment by skonto** [comment]
@mgencur gentle ping.

**Pull request merged**